rpc/server.go

```diff
@@ -161,7 +244,7 @@ func (s *Server) serveRequest(codec ServerCodec, singleShot bool, options CodecO
 		return
 	}()
 
-	ctx, cancel := context.WithCancel(context.Background())
+	ctx, cancel := context.WithCancel(context.WithValue(context.Background(), "codec", codec))
```

Comment:
Look for how `notifierKey{}` is used to add the notifier to the context. You want to take a look at `func NotifierFromContext(ctx context.Context) (Notifier, bool)` as well. The same holds for "id" that is added to the context.